lib/core-schema-types.js

```diff
@@ -400,10 +400,12 @@ class SchemaTypeMap extends SchemaType {
 		return (objtools.isPlainObject(value) ? 1 : 0);
 	}
 
-	toJSONSchema() {
-		return { type: 'object' };
+	toJSONSchema(subschema) {
+		return {
+			type: 'object',
+			properties: subschema.values
```

Comment:
`toJSONSchema()` will have to be called recursively on the subschemas here, like in SchemaTypeArray and SchemaTypeObject.  Otherwise, the subschemas won't get converted to JSON schema format.